cdap-app-templates/cdap-etl/cdap-etl-tools/src/main/java/co/cask/cdap/etl/tool/UpgradeTool.java

```diff
@@ -174,8 +174,15 @@ private boolean upgradePluginTemplates(JsonObject consoleData) {
         for (Map.Entry<String, Map<String, Map<String, PluginTemplate>>> appEntry :
           namespaceEntry.getValue().entrySet()) {
           String appName = appEntry.getKey();
-          PluginArtifactFinder pluginArtifactFinder =
-            appName.equals(BATCH_NAME) ? batchPluginArtifactFinder : realtimePluginArtifactFinder;
+          PluginArtifactFinder pluginArtifactFinder;
+          if (BATCH_NAME.equals(appName)) {
+            pluginArtifactFinder = batchPluginArtifactFinder;
+          } else if (REALTIME_NAME.equals(appName)) {
+            pluginArtifactFinder = realtimePluginArtifactFinder;
+          } else {
+            // don't know what this is, ignore it.
```

Comment:
ok, but wouldn't a warn message like 'skipped upgrade of a pipeline X' be informative - will help in case they're unable to make some pipeline work?
Or is this `else` case only triggered if there is some non-pipeline thing in the configuration store?